### PR TITLE
Added support for deeplinks

### PR DESCRIPTION
**Pull request by wadeking98**
Added support for deeplinks in the demo.
Everything except the OOB proof request working (not supported in wallet yet)

### CODE REVIEW

File: server/src/controllers/DeepLinkController.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        const state = await this.waitUntilConnected(params.connectionId)
+        if (state === 'complete' || state === 'responded') {
+            const resp = await apiCall.post('/proofs/request-proof', params)
+            return resp.data
+        }
+    }
+
+    async waitUntilConnected(connectionId: string): Promise<string> {
+        let state = ""
+        for (let i = 0; i < 10 && state !== 'complete' && state !== 'responded'; i++) {
```
Comment (amanji) [inline server/src/controllers/DeepLinkController.ts:38]: What does the for loop do?

Comment (wadeking98) [reply server/src/controllers/DeepLinkController.ts:38]: checks to see if the connection is complete once per second over a 10 second interval

Comment (wadeking98) [reply server/src/controllers/DeepLinkController.ts:38]: Right but setInterval is non-blocking so the function would immediately return

Comment (amanji) [reply server/src/controllers/DeepLinkController.ts:38]: So what happens if the connection doesn't take place within the 10 seconds?

Comment (wadeking98) [reply server/src/controllers/DeepLinkController.ts:38]: Then it returns the state as `invited` and exists. I figured that would be better than hanging forever if the connection never completes.

**Review by amanji — APPROVED** [review]

**Pull request merged**